### PR TITLE
quotemark: Add 'no-template' option

### PR DESCRIPTION
**Pull request by andy-hanson**
#### PR checklist

- [ ] Addresses an existing issue: #0000
- [X] New feature, bugfix, or enhancement
  - [X] Includes tests
- [X] Documentation update

#### Overview of change:

Adds the `no-template` option, which forbids *single-line* template literals with no interpolation.
Still allows tagged tempates and templates with interpolation.
Allows them if they contain *both* `"` and `'`, otherwise corrects to some kind of quote.

#### CHANGELOG.md entry:

[new-rule-option] `quotemark`: Add `no-template` option

### CODE REVIEW

File: src/configs/all.ts [hunk 1]
```diff
@@ -204,7 +204,12 @@ export const rules = {
     "prefer-method-signature": true,
     "prefer-switch": true,
     "prefer-template": true,
-    "quotemark": [true, "double", "avoid-escape"],
+    "quotemark": [
+        true,
+        "double",
+        "avoid-escape",
+        "no-template",
```
Comment (nchen63) [inline src/configs/all.ts]: rename to `avoid-template` since some templates are allowed

File: src/rules/quotemarkRule.ts [hunk 2]
```diff
@@ -44,6 +47,7 @@ export class Rule extends Lint.Rules.AbstractRule {
             * \`"${OPTION_DOUBLE}"\` enforces double quotes.
             * \`"${OPTION_JSX_SINGLE}"\` enforces single quotes for JSX attributes.
             * \`"${OPTION_JSX_DOUBLE}"\` enforces double quotes for JSX attributes.
+            * \`"${OPTION_NO_TEMPLATE}"\` forbids single-line template strings that do not contain string interpolations.
```
Comment (nchen63) [inline src/rules/quotemarkRule.ts]: also "untagged"

File: src/rules/quotemarkRule.ts [hunk 3]
```diff
@@ -75,39 +79,68 @@ export class Rule extends Lint.Rules.AbstractRule {
...
+            return ctx.addFailure(start, node.end, Rule.FAILURE_STRING(actualQuoteMark, fixQuoteMark),
+                new Lint.Replacement(start, node.end - start, fixQuoteMark + text + fixQuoteMark));
         }
-        return ts.forEachChild(node, cb);
+        ts.forEachChild(node, cb);
     });
 }
+
+function isMultiLine(node: ts.Node, sourceFile: ts.SourceFile): boolean {
+    return sourceFile.getLineAndCharacterOfPosition(node.getStart(sourceFile)).line !==
```
Comment (ajafff) [inline src/rules/quotemarkRule.ts]: you can avoid that boilerplate code by using `tsutils.isSameLine`
```ts
return !isSameLine(sourceFile, node.getStart(sourceFile), node.end);
```

File: src/rules/quotemarkRule.ts [hunk 4]
```diff
@@ -75,39 +79,68 @@ export class Rule extends Lint.Rules.AbstractRule {
...
 
 function walk(ctx: Lint.WalkContext<Options>) {
-    return ts.forEachChild(ctx.sourceFile, function cb(node: ts.Node): void {
-        if (node.kind === ts.SyntaxKind.StringLiteral) {
-            const expectedQuoteMark = node.parent!.kind === ts.SyntaxKind.JsxAttribute ? ctx.options.jsxQuoteMark : ctx.options.quoteMark;
-            const actualQuoteMark = ctx.sourceFile.text[node.end - 1];
+    const { sourceFile, options } = ctx;
+    ts.forEachChild(sourceFile, function cb(node) {
+        if (isStringLiteral(node)
+                || isNoSubstitutionTemplateLiteral(node)
```
Comment (ajafff) [inline src/rules/quotemarkRule.ts]: needs to check if the option is set

**Pull request merged**